### PR TITLE
Add retry to server requests

### PR DESCRIPTION
**Pull request by jp-ryuji**
# smoke test passed on local

```sh
$ grep 'const baseUrl' e2e/production.spec.ts
// const baseUrl = 'https://steexp.com'
const baseUrl = 'http://publicnet.local:3000' // for test on local
```

```sh
$ npm run test:e2e

> stellarexplorer@2.4.0 test:e2e
> playwright test


Running 11 tests using 4 workers
  11 passed (20.3s)
```

### CODE REVIEW

File: app/lib/stellar/server_request_utils.ts [hunk 1]
```diff
@@ -35,7 +35,24 @@ interface PageOptions {
...
+  return pRetry(run, {
+    onFailedAttempt: (error) => {
+      console.log(
+        `Attempt ${error.attemptNumber} failed. There are ${error.retriesLeft} retries left.`,
+      )
+    },
+    retries: 1,
+    factor: 1,
+    minTimeout: 1000,
+    maxTimeout: 2000,
```
[the remark this answers is not in the document]
Comment (chatch) [reply app/lib/stellar/server_request_utils.ts:51]: ok let's go with one, will see the impact that has on sentry and how many failures on the second try we still get. can consider a second retry after that ..

is there a pause in between? or just retry straight after the error is received?

[the remark this answers is not in the document]
Comment (chatch) [reply app/lib/stellar/server_request_utils.ts:51]: how slow was a slow response? like 5 seconds or more?

**Review by chatch — APPROVED** [review]
looks good, happy to merge this if you are.  seems the module stuff is basically fine as it is now right?

good to get this in the wild and see how the retries go and then potentially tweak after that.

File: app/lib/stellar/server_request_utils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jp-ryuji) [reply app/lib/stellar/server_request_utils.ts:51]: > is there a pause in between? or just retry straight after the error is received?

Yes.
The options work as follows:

- `factor`: This is a factor for calculating the retry interval. The retry interval is the previous retry interval multiplied by `factor`. If `factor` is set to 1, each retry interval will be constant (between `minTimeout` and `maxTimeout`).

- `minTimeout` and `maxTimeout`: These specify the minimum and maximum retry interval in milliseconds. The retry interval will be a value between `minTimeout` and `maxTimeout`. If `factor` is set to 1, the retry interval will be constant between these values.

So there is an interval between 1000 and 2000 milliseconds with the current configuration.
If the `retries` is set to 2, I think the first and second intervals are 1000 and 2000 ms respectively.

> how slow was a slow response? like 5 seconds or more?

It says 40459.112 ms.
I'm not very sure if it took that long, but it felt like 10 seconds or so. 

```
GET /account/GCYBS25RYBGG7EDKP5FBDJIYFPSGKQNOIMRIJUTQXJSBBGHXPSH7RZB6/payments?_data=routes%2Faccount.%24accountId.payments 500 - - 40459.112 ms
```

**Comment by jp-ryuji** [comment]
> seems the module stuff is basically fine as it is now right?

Yes.
The current config, `"commonjs"`, is fine.

File: app/lib/stellar/server_request_utils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jp-ryuji) [reply app/lib/stellar/server_request_utils.ts:51]: It says 13435.425 ms in a success case.

```
GET /account/GCLM2V4D45RTGXTUYB6LU32Q3IUWZC5NN4KF5S6SCYXEZZLUOIKDKCJI/trades?_data=routes%2Faccount.%24accountId.trades 200 - - 13435.425 ms
```

**Comment by beber110** [comment]
**
### 
@**

**Pull request merged**